posthog-core/src/index.ts

```diff
@@ -202,27 +206,27 @@ export abstract class PostHogCore {
         event: '$identify',
         properties: {
           ...(properties || {}),
-          $anon_distinct_id: this.getDistinctId(), // TODO: Should this be the previous distinct id or the original anon id??
+          $anon_distinct_id: this.getAnonymousId(),
```

Comment:
This should always be the previous distinctID?

As we talked earlier, if people forget to set `reset()` and we call `identify()` with the anon_distinct_id set to an anon ID, then both the distinctIDs supplied by the user will be merged, which isn't something we want.

If anon_distinct_id is an existing identified id though, they're not merged together (you need an `alias` call to do this explicitly).

Reply:
As you saw in the other comment, this is exactly what essentially happens as the anonId is overwritten with the old distinctId.